From 6dd39ec9d4dcbe900eeb98bf6654ff2fcc3eae84 Mon Sep 17 00:00:00 2001
From: Saurabh Sengar <saurabh.singh@xilinx.com>
Date: Tue, 6 Mar 2018 16:20:13 +0530
Subject: [PATCH 0327/1851] drm: xlnx: mixer: reset the mixer IP before
 programming

commit 79e6fec89cf946d02e7dc43621b6c1c149c241c3 from
https://github.com/Xilinx/linux-xlnx.git

Resetting the HLS IP as soon as gpio driver is probed.

Signed-off-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_mixer.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_mixer.c b/drivers/gpu/drm/xlnx/xlnx_mixer.c
index d043980f53ee..23045413c39f 100644
--- a/drivers/gpu/drm/xlnx/xlnx_mixer.c
+++ b/drivers/gpu/drm/xlnx/xlnx_mixer.c
@@ -1896,6 +1896,8 @@ static int xlnx_mix_dt_parse(struct device *dev, struct xlnx_mix *mixer)
 			dev_err(dev, "No reset gpio info from dts for mixer\n");
 		return ret;
 	}
+	gpiod_set_raw_value(mixer_hw->reset_gpio, 0);
+	gpiod_set_raw_value(mixer_hw->reset_gpio, 1);
 
 	ret = of_address_to_resource(node, 0, &res);
 	if (ret) {
-- 
2.31.1

